Assets/Teleporter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Teleporter : MonoBehaviour {
+
+	Transform _targetTransform = null;
+	[SerializeField] string _name = "";
+	[SerializeField] AudioSource _teleportSound = null;
+
+	void Start () {
+		_targetTransform = GetComponentInChildren<TeleportTargetTag>().gameObject.GetComponent<Transform>();
+
+		TextMesh[] textMeshes =  GetComponentsInChildren<TextMesh>();
+
+		foreach ( TextMesh textMesh in textMeshes )
+		{
+			textMesh.text =_name;
+		}
+	}
+
+	void OnTriggerEnter( Collider col )
+	{
+		if ( col.gameObject.transform.parent == null )
+		{
+			return;
+		}
+
+		// Get to the player from the bumper
+		PlayerControls controls = col.gameObject.GetComponent<Transform>().parent.gameObject.GetComponent<PlayerControls>();
```

Comment:
You could use [`GetComponentInParent()`](http://docs.unity3d.com/ScriptReference/GameObject.GetComponentsInParent.html) instead of hunting down the transform.